onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -2656,6 +2702,34 @@ including arg name, arg type (contains both type and shape).)pbdoc")
             auto res = sess->GetSessionHandle()->GetModelMetadata();
             OrtPybindThrowIfError(res.first);
             return *(res.second); }, py::return_value_policy::reference_internal)
+      .def_property_readonly("inputs_meminfo", [](const PyInferenceSession* sess) -> std::vector<const OrtMemoryInfo*> { 
+          Ort::ConstSession session(reinterpret_cast<const OrtSession*>(sess->GetSessionHandle()));
+          auto inputs_mem_info = session.GetMemoryInfoForInputs();
+          std::vector<const OrtMemoryInfo*> result;
+          result.reserve(inputs_mem_info.size());
+          for (const auto& info : inputs_mem_info) {
+            result.push_back(info);
+          }
+          return result; }, py::return_value_policy::reference_internal)
```

Comment:
I don't think `reference_internal` will works as expected because this is returning a `std::vector<>` by value. The [pybind documentation](https://pybind11.readthedocs.io/en/stable/advanced/functions.html#return-value-policies) states that it will become `return_value_policy::move`:

> If the return value is an lvalue reference or a pointer, the parent object (the implicit this, or self argument of the called method or property) is kept alive for at least the lifespan of the return value. **Otherwise this policy falls back to return_value_policy::move**